extensions/debug-server-ready/src/extension.ts

```diff
@@ -246,6 +251,38 @@ class ServerReadyDetector extends vscode.Disposable {
 		});
 	}
 
+	private async startConfigDebugSession(session: vscode.DebugSession, config: vscode.DebugConfiguration) {
```

Comment:
Any way to share logic with `startNamedDebugSession`?

Reply:
I had one throught:

```
	private async startConfigDebugSession(session: vscode.DebugSession, name: string, config?: vscode.DebugConfiguration) {
		const args = session.configuration.serverReadyAction as ServerReadyAction;
		if (!args.killOnServerStop) {
			await vscode.debug.startDebugging(session.workspaceFolder, config ?? name);
			return;
		}

		const cts = new vscode.CancellationTokenSource();
		const newSessionPromise = this.catchStartedDebugSession(x => x.name === name, cts.token);

		if (!await vscode.debug.startDebugging(session.workspaceFolder, config ?? name)) {
			cts.cancel();
			cts.dispose();
			return;
		}

		const createdSession = await newSessionPromise;
		cts.dispose();

		if (!createdSession) {
			return;
		}

		const stopListener = this.onDidSessionStop(async () => {
			stopListener.dispose();
			this.disposables.delete(stopListener);
			await vscode.debug.stopDebugging(createdSession);
		});
		this.disposables.add(stopListener);
	}
```

Thoughts?

Reply:
Fine with me